docs/src/type_stability.md

```diff
@@ -1,3 +1,219 @@
 # [The Importance of Type-Stability](@id doc:Type-Stability)
 
-This page is still under construction.
+You are here because you have probably heard about the excellent performance of Julia compared to other common programming languages like Python. One of the reasons is the Just-In-Time (JIT) compiler of Julia, which is able to generate highly optimized machine code. However, the JIT compiler can only do its job if the code type can be inferred. What does this mean exactly? Here we try to explain it in a simple way.
```

Comment:
I think it would be better to also mention julia official document page here.
like:

~~What does this mean exactly? Here we try to explain it in a simple way.~~
You can also read the section [Performance Tips](https://docs.julialang.org/en/v1/manual/performance-tips/#man-performance-tips) in Julia's official documentation for more details.

Here, we try to explain it briefly.